client/signup/steps/domain-upsell/index.jsx

```diff
@@ -51,17 +47,19 @@ export default function DomainUpsellStep( props ) {
 		}
 	);
 
+	const [ isLoadingExperimentAssignment, experimentAssignment ] = useExperiment(
+		'domain_upsell_emphasize_free_v3'
+	);
+
 	const isInDomainUpsellEmphasizeFreeTest =
-		'treatment' ===
-		useSelector( ( state ) => getVariationForUser( state, 'domain_upsell_emphasize_free_v2' ) );
+		experimentAssignment && 'treatment' === experimentAssignment.variationName;
```

Comment:
Tip: optional chaining is useful here: `'treatment' === experimentAssignment?.variationName;`